oak-core/src/main/java/org/apache/jackrabbit/oak/Oak.java

```diff
@@ -576,6 +577,10 @@ public Oak with(@NotNull Whiteboard whiteboard) {
             newSettings.setLimitReads(this.queryEngineSettings.getLimitReads());
             this.queryEngineSettings = new AnnotatedQueryEngineSettings(newSettings);
         }
+        if (queryEngineSettings != null) {
+            Feature prefetchFeature = Feature.newFeature(QueryEngineSettings.FT_NAME_PREFETCH_FOR_QUERIES, whiteboard);
```

Comment:
Should register feature with closer to make sure it is removed from whiteboard when the ContentRepository is closed.
```suggestion
            Feature prefetchFeature = Feature.newFeature(QueryEngineSettings.FT_NAME_PREFETCH_FOR_QUERIES, whiteboard);
            closer.register(prefetchFeature);
```